src/csharp/Intel/Generator/Extended/CSharp/CSharpExtendedSyntaxGenerator.cs

```diff
@@ -0,0 +1,473 @@
+										writer.WriteLine($"op = Code.{items[0].Code.Name(Converter)};");
+									}
+									else {
+										bool isFirst = true;
+										bool has64 = false;
+										for (int i = items.Count - 1; i >=0; i--) {
+											var opcode = items[i];
+											if (opcode is LegacyOpCodeInfo legacy) {
+												if (!isFirst) writer.Write(" else ");
+												isFirst = false;
+												if (i == 0) {
+													writer.WriteLine("{");
+												}
+												else {
+													var bitness = legacy.OperandSize == OperandSize.Size16 ||
+													              legacy.AddressSize == AddressSize.Size16 ||
+														          legacy.Flags == OpCodeFlags.Mode16 ? 16 : legacy.OperandSize == OperandSize.Size32  || 
+														                                                       legacy.AddressSize == AddressSize.Size32  || 
+														                                                       legacy.Flags == (OpCodeFlags.Mode16 | OpCodeFlags.Mode32) ? 32 : 64;
+													if (bitness == 64) has64 = true;
+													writer.WriteLine($"if (Bitness {(has64 ? "==" : ">=")} {bitness}) {{");
+												}
+												using (writer.Indent()) {
+													writer.WriteLine($"op = Code.{opcode.Code.Name(Converter)};");
+												}
+												writer.Write("}");
+											}
+										}
+										writer.WriteLine();
+									}
+								}
+
+								writer.Write("AddInstruction(Instruction.Create(op");
+
+								for (var i = 0; i < renderArgs.Count; i++) {
+									var renderArg = renderArgs[i];
+									if (renderArg.Kind == ArgKind.HiddenMemory) continue;
+									writer.Write(", ");
+									writer.Write(renderArg.Name);
+								}
+
+								writer.WriteLine("));");
+							}
+							writer.WriteLine("}");
+						}
+					}
+					writer.WriteLine("}");
+				}
+				writer.WriteLine("}");
+				writer.WriteLine("#endif");
+			}
+		}
+
+		private struct RenderArg {
+			public RenderArg(string name, string type, ArgKind kind) {
+				Name = name;
+				Type = type;
+				Kind = kind;
+			}
+
+			public string Name;
+
+			public string Type;
+
+			public ArgKind Kind;
+		}
+
+		void GenerateOpCodeSelectorFromRegisterOrMemory(FileWriter writer, OpCodeInfoGroup group, List<RenderArg> args, int argIndex, string methodName) {
+			bool hasIf = false;
+			FileWriter.Indenter indenter = default;
+			if (group.ItemsWithImmediateByte1 != null) {
+				var immReg = args[group.ImmediateArgIndex];
+				writer.WriteLine($"if ({immReg.Name} == 1) {{");
+				hasIf = true;
+				using (writer.Indent()) {
+					GenerateOpCodeSelectorFromRegisterOrMemory(writer, group, group.ItemsWithImmediateByte1, args, argIndex, methodName);
+				}
+
+				writer.WriteLine("}");
+			}
+
+			if (group.ItemsWithImmediateByte != null) {
+				if (hasIf) {
+					writer.Write("else ");
+				}
+
+				if (group.Items.Count > 0) {
+					var immReg = args[group.ImmediateArgIndex];
+					writer.WriteLine($"if ((uint){immReg.Name} <= byte.MaxValue) {{");
```

Comment:
It fits in a byte, so `-128 .. 255` should be valid values, eg.:

sbyte.MinValue <= imm && imm <= byte.MaxValue

Reply:
That's what `(uint)imm <= byte.MaxValue` is covering and it performs this with one comparison instead of 2.

Reply:
`adc al,-128` and `adc al,255` should both be accepted, but the uint cast prevents the `-128` constant from using the 1-byte imm opcode.

If it's sign extended from a byte to a word/dword/qword, we must only allow -128..127 .

- byte: verify that it's -128..255
- sign extended to word/dword/qword: verify that it's -128..127

Reply:
Oops, sorry, you are right, the cast will convert -1 to a value that is beyond 255. I will fix this.